vnext/ReactUWP/Views/ScrollViewManager.h

```diff
@@ -18,23 +21,34 @@ class ScrollViewManager : public ControlViewManager
   folly::dynamic GetNativeProps() const override;
   folly::dynamic GetExportedCustomDirectEventTypeConstants() const override;
 
+  facebook::react::ShadowNode* createShadow() const override;
+
   void AddView(XamlView parent, XamlView child, int64_t index) override;
   void RemoveAllChildren(XamlView parent) override;
   void RemoveChildAt(XamlView parent, int64_t index) override;
 
-  void UpdateProperties(ShadowNodeBase* nodeToUpdate, folly::dynamic reactDiffMap) override;
-  void DispatchCommand(XamlView viewToUpdate, int64_t commandId, const folly::dynamic& commandArgs) override;
+  void SnapToInterval(XamlView parent, float interval);
+  void SnapToOffsets(XamlView parent, const winrt::IVectorView<float>& offsets);
 
 protected:
   XamlView CreateViewCore(int64_t tag) override;
 
 private:
-  void AddHandlers(winrt::Windows::UI::Xaml::Controls::ScrollViewer& scrollViewer, int64_t tag);
+  void AddHandlers(const winrt::ScrollViewer& scrollViewer, int64_t tag);
   void EmitScrollEvent(
-    winrt::Windows::UI::Xaml::Controls::ScrollViewer& scrollViewer,
+    const winrt::ScrollViewer& scrollViewer,
     int64_t tag,
     const char* eventName,
     double x, double y, double zoom);
+
+  std::map<int64_t, winrt::ScrollViewer::ViewChanging_revoker> m_scrollViewerViewChangingRevokers{};
+  std::map<int64_t, winrt::ScrollViewer::DirectManipulationCompleted_revoker> m_scrollViewerDirectManipulationCompletedRevokers{};
+  std::map<int64_t, winrt::ScrollViewer::DirectManipulationStarted_revoker> m_scrollViewerDirectManipulationStartedRevokers{};
+
+  std::map<int64_t, bool> m_isScrollingFromInertia{};
+  std::map<int64_t, bool> m_isScrolling{};
+
```

Comment:
at some point we should move all of these members to the shadownode and not have maps, just members on each shadow node.

Reply:
I was trying to figure out how to do this actually, and was going to reach out. I need these properties in order to raise the appropriate RN events via EmitScrollEvent. I had them on ShadowNode but couldn't figure out the appropriate way to get the value off the shadow node during this event callback. Alternatively we could raise the event from the shadow node and do the event handling in that object but with that I couldn't figure out how to get the ReactInstance from the shadow node.

Reply:
check out PickerViewManager.cpp for an example of registering and firing events from a shadow node